var/spack/repos/builtin/packages/rust/package.py

```diff
@@ -103,10 +103,17 @@ class Rust(Package):
     phases = ["configure", "build", "install"]
 
     @classmethod
-    def determine_version(csl, exe):
-        output = Executable(exe)("--version", output=str, error=str)
-        match = re.match(r"rustc (\S+)", output)
-        return match.group(1) if match else None
+    def determine_spec_details(cls, prefix, exes_in_prefix):
+        rustc_candidates = list(x for x in exes_in_prefix if os.path.basename(x) == "rustc")
+        cargo_candidates = list(x for x in exes_in_prefix if os.path.basename(x) == "cargo")
```

Comment:
```suggestion
        rustc_candidates = [x for x in exes_in_prefix if os.path.basename(x) == "rustc"]
        cargo_candidates = [x for x in exes_in_prefix if os.path.basename(x) == "cargo"]
```

Quick suggestion to simplify the syntax in the list comprehension.

Reply:
I was wondering why that syntax is used in https://spack.readthedocs.io/en/latest/packaging_guide.html#custom-detection-workflow... We should update it there too then. Is this a remnant of old python syntax?